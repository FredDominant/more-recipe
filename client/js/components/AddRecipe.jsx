import React from 'react';
import { connect } from 'react-redux';

import uploadImage from '../utils/uploadImage';
import recipeValidator from '../validation/recipeValidator';
/**
 * @description componet holds form to add recipes
 *
 * @class AddRecipe
 * @extends {React.Component}
 */
class AddRecipe extends React.Component {
/**
 * Creates an instance of AddRecipe.
 * @param {any} props
 * @memberof AddRecipe
 */
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      description: '',
      ingredients: '',
      directions: '',
      recipeImage: '',
      errors: {}
    };
    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onUpload = this.onUpload.bind(this);
  }
  /**
   *
   * @returns {null} null
   * @param {any} event
   * @memberof AddRecipe
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  /**
 *
 * @returns {null} null
 * @param {any} event
 * @memberof AddRecipe
 */
  onUpload(event) {
    const image = event.target.files[0];
    this.setState({ recipeImage: image });
  }
  /**
 *
 * @returns {null} null
 * @memberof AddRecipe
 */
  isValid() {
    const { errors, isValid } = recipeValidator(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }
  /**
   *
   * @returns {null} null
   * @memberof AddRecipe
   */
  addRecipe() {
    uploadImage(this.state.recipeImage)
      .then((response) => {
        // console.log('image url after axios is', response.data.secure_url);
        this.setState({ recipeImage: response.data.secure_url });
        // console.log('image url after updating state is', this.state.recipeImage);
        // dispatch this to store;
      })
      .catch((error) => {
        console.log(error);
      });
  }
  /**
 *
 * @returns {null} null
 * @param {any} event
 * @memberof AddRecipe
 */
  handleSubmit(event) {
    event.preventDefault();
    if (this.isValid()) {
      this.addRecipe();
    }
  }
  /**
   *
   * @returns {HTML} html
   * @memberof AddRecipe
   */
  render() {
    const errors = this.state.errors;
    return (
      <div>
        <div className="modal fade" id="addRecipe" tabIndex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
          <div className="modal-dialog" role="document">
            <div className="modal-content container">
              <div className="modal-header">
                <h5 className="modal-title"><span id="title"><h3 title="More Recipes and cooking tips">More Recipes</h3></span></h5>
                <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                  <span aria-hidden="true">&times;</span>
                </button>
              </div>
              <div className="modal-body">
                <div className="container error-body">
                  {errors.name && <div className="alert alert-dismissible alert-danger" role="alert">{errors.name}</div>}
                  {errors.description && <div className="alert alert-dismissible alert-danger" role="alert">{errors.description}</div>}
                  {errors.ingredients && <div className="alert alert-dismissible alert-danger" role="alert">{errors.ingredients}</div>}
                  {errors.directions && <div className="alert alert-dismissible alert-danger" role="alert">{errors.directions}</div>}
                </div>
                <form onSubmit={this.handleSubmit}>
                  <div className="form-group">
                    <label htmlFor="exampleInputEmail1">Recipe Name</label>
                    <input
                      type="text"
                      className="form-control"
                      name="name"
                      id="recipeName"
                      aria-describedby="small-recipe-name"
                      placeholder="Awesome Recipe!"
                      value={this.state.name}
                      onChange={this.onChange}
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="recipe-description">Recipe Description:</label>
                    <input
                      type="text"
                      className="form-control"
                      name="description"
                      id="recipe-description"
                      placeholder="Awesome sauce made out of nothing!"
                      value={this.state.description}
                      onChange={this.onChange}
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="cook-directions">Ingredients:</label>
                    <textarea
                      className="form-control"
                      id="cook-directions"
                      name="ingredients"
                      rows="3"
                      placeholder="Onions, tomatoes, grilled chicken, nutmeg"
                      value={this.state.ingredients}
                      onChange={this.onChange}
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="cook-directions">Directions to cook:</label>
                    <textarea
                      className="form-control"
                      name="directions"
                      id="cook-directionss"
                      rows="3"
                      placeholder="dice tomatoes, tear chicken, boil onions, mix with nutmeg"
                      value={this.state.directions}
                      onChange={this.onChange}
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="recipeImage">Recipe Image</label>
                    <input
                      type="file"
                      className="form-control-file"
                      name="file"
                      id="recipeImage"
                      onChange={this.onUpload}
                    />
                  </div>
                  <button className="btn btn-primary">Add Recipe</button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({

});
export default connect(mapStateToProps)(AddRecipe);
