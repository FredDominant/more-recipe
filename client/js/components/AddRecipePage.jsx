import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import MDSpinner from 'react-md-spinner';
import toastr from 'toastr';

import uploadImage from '../utils/uploadImage';
import recipeValidator from '../validation/recipeValidator';
import addRecipe from '../actions/addRecipe';
/**
 *
 * @class AddRecipePage
 *
 * @extends {React.Component}
 */
export class AddRecipePage extends React.Component {
/**
 * @description Creates an instance of AddRecipePage.
 *
 * @param {any} props
 *
 * @memberof AddRecipePage
 */
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      description: '',
      ingredients: '',
      directions: '',
      isUploading: false,
      recipeImage: '/images/noImage.png',
      errors: {},
      addRecipeError: '',
      uploadImageError: ''
    };
    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onUpload = this.onUpload.bind(this);
    this.handleSelectImage = this.handleSelectImage.bind(this);
  }
  /**
 * @description component lifecycle method
 *
 * @returns {null} null
 *
 * @param {nextProp} nextProp object from store
 *
 * @memberof AddRecipe
 */
  componentWillReceiveProps(nextProp) {
    this.setState({ addRecipeError: nextProp.errorMessage });
  }

  /**
 *
 * @returns {null} null
 * @param {any} event
 * @memberof AddRecipe
 */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  /**
*
* @returns {null} null

* @param {any} event

* @memberof AddRecipe
*/
  onUpload(event) {
    const reader = new FileReader();
    reader.onload = (e) => {
      this.setState({ recipeImage: e.target.result });
    };
    reader.readAsDataURL(event.target.files[0]);
  }
  /**
*
* @returns {null} null

* @memberof AddRecipe
*/
  isValid() {
    const { errors, isValid } = recipeValidator(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }
  /**
 *
 * @returns {null} null
 *
 * @memberof AddRecipe
 */
  addNewRecipe() {
    const { recipeImage, name, description, directions, ingredients } = this.state;
    if (recipeImage !== '/images/noImage.png') {
      this.setState({ isUploading: true });
      return uploadImage(recipeImage)
        .then((response) => {
          const recipeUrl = response.data.secure_url;
          console.log('recipe url is', recipeUrl);
          this.setState({ recipeImage: recipeUrl, isUploading: false });
          this.props.createRecipe(this.state);
        })
        .catch(() => {
          this.setState({ isUploading: false, uploadImageError: 'Could not upload image. Recipe not added' });
          toastr.options = {
            closeButton: true
          };
          toastr.error(this.state.uploadImageError);
        });
    }
    console.log('jumped to this place');
    this.props.createRecipe({ name, directions, ingredients, description });
  }
  /**
*
* @returns {null} null

* @param {any} event

* @memberof AddRecipe
*/
  handleSubmit(event) {
    event.preventDefault();
    if (this.isValid()) {
      this.addNewRecipe();
    }
  }
  /**
   *
   * @memberof AddRecipePage
   *
   * @returns {null} null
   */
  handleSelectImage() {
    $('input[type=file]').click();
  }
  /**
   *
   * @returns {null} null
   *
   * @memberof AddRecipePage
   */
  render() {
    const { errors, recipeImage, name, description, directions, ingredients } = this.state;
    return (
      <div className="container-fluid">
        <div id="add-recipe-form" className="container">
          <form onSubmit={this.handleSubmit}>
            <div className="row">
              <div className="col-sm-4">
                <div
                  className="recipe-image"
                  id="add-recipe-image-container"
                  onClick={this.handleSelectImage}
                  role="presentation"
                >
                  <img
                    className="img-thumbnail"
                    src={recipeImage}
                    alt=""
                    srcSet=""
                    id="add-recipe-image"
                  />
                </div>
                <br />
                <input
                  style={{ display: 'none' }}
                  type="file"
                  name="file"
                  id="file-upload"
                  onChange={this.onUpload}
                />
              </div>
              <div className="col-sm-8">
                <div className="form-group">
                  <label htmlFor="recipeName">Recipe Name</label>
                  <input
                    type="text"
                    id="recipeName"
                    className="form-control"
                    name="name"
                    value={name}
                    onChange={this.onChange}
                  />
                  {errors.name && <small className="form-text text-muted">
                    <span className="error-text"> {errors.name} </span>
                  </small>}
                </div>

                <div className="form-group">
                  <label htmlFor="recipeDescription">Recipe Description</label>
                  <input
                    type="text"
                    id="recipeDescription"
                    className="form-control"
                    name="description"
                    value={description}
                    onChange={this.onChange}
                  />
                  {errors.description && <small className="form-text text-muted">
                    <span className="error-text"> {errors.description} </span>
                  </small>}
                </div>

                <div className="form-group">
                  <label htmlFor="recipeDirections">Recipe Directions</label>
                  <textarea
                    type="text"
                    rows="5"
                    id="recipeDirections"
                    className="form-control"
                    name="directions"
                    value={directions}
                    onChange={this.onChange}
                  />
                  {errors.directions && <small className="form-text text-muted">
                    <span className="error-text"> {errors.directions} </span>
                  </small>}
                </div>

                <div className="form-group">
                  <label htmlFor="recipeingredients">Ingredients</label>
                  <textarea
                    type="text"
                    rows="5"
                    id="recipeIngredients"
                    className="form-control"
                    placeholder="Comma separated list of ingredients"
                    name="ingredients"
                    value={ingredients}
                    onChange={this.onChange}
                  />
                  {errors.ingredients && <small className="form-text text-muted">
                    <span className="error-text"> {errors.ingredients} </span>
                  </small>}
                </div>

                <div className="form-group">
                  <button className="btn btn-primary">
                    {
                      !this.state.isUploading &&
                      <h6 className="text-center"> Add Recipe </h6>
                    }
                    {
                      this.state.isUploading &&
                      <h6 className="text-center"> Uploading Image... </h6>
                    }
                    {
                      this.props.loading &&
                      <span> <MDSpinner /></span>
                    }
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  errorMessage: state.addRecipe.addRecipeErrorMessage,
  addRecipeSuccess: state.addRecipe.addRecipeSuccess,
  loading: state.isUploading
});

const mapDispatchToProps = dispatch => ({
  createRecipe: recipe => dispatch(addRecipe(recipe))
});

AddRecipePage.propTypes = {
  createRecipe: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired
};
AddRecipePage.defaultProps = {
  errorMessage: null,
  addRecipeSuccess: null
};
export default connect(mapStateToProps, mapDispatchToProps)(AddRecipePage);

