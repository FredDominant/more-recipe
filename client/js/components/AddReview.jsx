import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import addReview from '../actions/addReview';
/**
 *
 * @class AddReview
 *
 * @extends {React.Component}
 */
export class AddReview extends React.Component {
/**
 * @description Creates an instance of AddReview.
 *
 * @param {object} props
 *
 * @memberof AddReview
 */
  constructor(props) {
    super(props);
    this.state = {
      content: ''
    };
    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  /**
   *
   * @param {object} event
   *
   * @memberof AddReview
   *
   * @returns {null} null
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  /**
   * @returns {null} null
   *
   * @param {object} event
   *
   * @memberof AddReview
   */
  handleSubmit(event) {
    const { content } = this.state;
    const recipeId = this.props.recipeId;
    event.preventDefault();
    if (content.trim().length) {
      this.props.postReview(content, recipeId);
      this.setState({ content: '' });
    }
  }
  /**
   *
   *
   * @memberof AddReview
   *
   * @return {ReactElement} markup
   */
  render() {
    return (
      <div>
        <div id="review-form">
          <div>
            <div>
              <form onSubmit={this.handleSubmit}>
                <input
                  type="text"
                  className="form-control"
                  name="content"
                  placeholder="Add a review..."
                  onChange={this.onChange}
                  value={this.state.content}
                  required
                />
                <br />
                <button className="btn" id="add-review-button">
                  <span><i className="fas fa-comment-alt" /> </span>
                  Post Review
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
AddReview.propTypes = {
  postReview: PropTypes.func.isRequired,
  recipeId: PropTypes.number
};
AddReview.defaultProps = {
  recipeId: 0
};
const mapDispatchToProps = dispatch => ({
  postReview: (review, recipeId) => dispatch(addReview(review, recipeId))
});
export default connect(null, mapDispatchToProps)(AddReview);

