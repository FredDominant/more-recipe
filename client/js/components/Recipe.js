import React from 'react';
import ReactDOM from 'react-dom';

import { Search } from './Search';
import { Navbar } from './Navbar';
import { Signup } from './Signup';
import { Login } from './Login';

export class Recipe extends React.Component {
  render() {
    return (
			<div>
				<Navbar user="registered"/>
				<div className="container-fluid">
					<h3>Recipe details page</h3>
				</div>
      </div>
    );
  }
}