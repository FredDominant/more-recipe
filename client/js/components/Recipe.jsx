import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Navbar from './Navbar';
import getOneRecipe from '../actions/getOneRecipe';
import upvoteRecipe from '../actions/upvote';
import addFavourites from '../actions/addFavourites';
import downvoteRecipe from '../actions/downvote';
import ViewReviews from '../components/ViewReviews';
import AddReview from '../components/AddReview';
import capitalize from '../utils/capitalize';
/**
 *
 *
 * @class Recipe
 * @extends {React.Component}
 */
class Recipe extends React.Component {
/**
 * Creates an instance of Recipe.
 * @param {any} props
 * @memberof Recipe
 */
  constructor(props) {
    super(props);
    this.state = {
      recipe: {},
      reviews: [],
      owner: ''
    };
    this.handleUpvote = this.handleUpvote.bind(this);
    this.handleDownvote = this.handleDownvote.bind(this);
    this.handleFavourite = this.handleFavourite.bind(this);
  }
  /**
   * @returns {null} null
   *
   * @param {any} nextProps
   * @memberof Recipe
   */
  componentWillMount() {
    this.props.getRecipeDetails(this.props.match.params.recipeId);
  }
  /**
   *
   * @return {null} null
   * @param {any} nextProps
   * @memberof Recipe
   */
  componentWillReceiveProps(nextProps) {
    const recipe = nextProps.recipe;
    const reviews = nextProps.recipe.Reviews;
    const owner = nextProps.recipe.User;
    this.setState({ recipe, reviews, owner });
  }

  /**
 * @description handles upvote event
 * @returns {null} null
 * @memberof Recipe
 */
  handleUpvote() {
    const recipeId = this.props.recipe.id;
    this.props.upvoteRecipe(recipeId);
  }

  /**
 * @description handles downvote event
 * @returns {null} null
 * @memberof Recipe
 */
  handleDownvote() {
    const recipeId = this.props.recipe.id;
    this.props.downvoteRecipe(recipeId);
  }

  /**
 * @description handles upvote event
 * @returns {null} null
 * @memberof Recipe
 */
  handleFavourite() {
    const recipeId = this.props.recipe.id;
    this.props.favourite(recipeId);
    console.log('favourited');
  }
  /**
   *
   *
   * @returns {component} react component
   * @memberof Recipe
   */
  render() {
    const ingredients = (this.state.recipe.ingredients) ? (this.state.recipe.ingredients) : '';
    console.log('ingredients is', ingredients);
    const splitIngredients = ingredients.split(',').map((item, i) => (
      <li key={`${i}`}>{item}</li>
    ));

    const { reviews } = this.state;
    const sortedReviews = reviews.sort((a, b) => (b.id - a.id));
    const allReviews = sortedReviews.map((review, i) => (
      <div key={`review ${i + 1}`} className="container">
        <ViewReviews
          image={review.User.picture}
          firstname={review.User.firstname}
          lastname={review.User.lastname}
          content={review.content}
        />
      </div>
    ));
    return (
      <div >
        <Navbar />
        <div className="container">
          <div className="container">
            <br />
            <div className="row">
              <div className="col-md-6 col-sm-8">
                <div className="image-container">
                  <img src={this.state.recipe.picture} alt="" height="50" width="50" className="img-thumbnail" />
                </div>
              </div>
              <div className="col-md-6">
                <h2>{this.state.recipe.name ? capitalize(this.state.recipe.name) : '' }</h2>
                <h5>{this.state.recipe.description ? capitalize(this.state.recipe.description) : '' }</h5>
                <ul>
                  {splitIngredients}
                </ul>
              </div>
            </div>
            <br />

            <h3>Ingredients: {this.state.recipe.ingredients}</h3>
            <h3>Recipe Owner: {`${this.state.owner.firstname} ${this.state.owner.lastname}`}</h3>
            <h3>Recipe Directions: {this.state.recipe.directions}</h3>
            <h3>Upvotes: {this.state.recipe.upvote}</h3>
            <h3>Downvotes: {this.state.recipe.downvote}</h3>
            <hr />
            {this.props.authenticated && <div className="actions">
              <div className="btn-group" role="group" >

                <button
                  type="button"
                  title="upvote this recipe"
                  className="btn btn-outline-danger"
                  onClick={this.handleUpvote}
                ><i className="far fa-thumbs-up" /></button>

                <button
                  type="button"
                  title="downvote this recipe"
                  className="btn btn-outline-danger"
                  onClick={this.handleDownvote}
                ><i className="far fa-thumbs-down" /></button>

                <button
                  type="button"
                  title="add to your favourites"
                  className="btn btn-outline-danger"
                  onClick={this.handleFavourite}
                ><i className="fab fa-gratipay" /></button>
              </div>
            </div> }
            <div className="container-fluid" id="review-body">
              <h5>Reviews</h5>
              <div className="add-review-form">
                {this.props.authenticated && <AddReview recipeId={this.state.recipe.id} />}
                <br />
              </div>
              { allReviews.length > 0 && <div>
                {allReviews}
              </div>}
              { !allReviews.length && <div className="emptyContent">
                <h2>There are currently no reviews for this recipe.</h2>
              </div>}
            </div>
          </div>
        </div>
      </div>
    );
  }
}
Recipe.propTypes = {
  match: PropTypes.shape().isRequired,
  getRecipeDetails: PropTypes.func.isRequired,
  upvoteRecipe: PropTypes.func.isRequired,
  downvoteRecipe: PropTypes.func.isRequired,
  favourite: PropTypes.func.isRequired,
  recipe: PropTypes.shape(),
  authenticated: PropTypes.bool.isRequired
};
Recipe.defaultProps = {
  recipe: {}
};

const mapStateToProps = state => ({
  recipe: state.getOneRecipe.singleRecipe,
  authenticated: state.auth.isAuthenticated
});

const mapDispatchToProps = dispatch => ({
  getRecipeDetails: recipeId => dispatch(getOneRecipe(recipeId)),
  upvoteRecipe: recipeId => dispatch(upvoteRecipe(recipeId)),
  downvoteRecipe: recipeId => dispatch(downvoteRecipe(recipeId)),
  favourite: recipeId => dispatch(addFavourites(recipeId))
});

export default connect(mapStateToProps, mapDispatchToProps)(Recipe);

