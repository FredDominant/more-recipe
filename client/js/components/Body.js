import React from 'react';
import ReactDOM from 'react-dom';
import { Switch, Route } from 'react-router-dom';

import { Home } from './Home';
import { Profile } from './Profile';
import { Recipe } from './Recipe'
import { UserHome } from './UserHome';

export class Body extends React.Component {
  render() {
    return (
			<Switch>
				<Route path='/' exact component={Home} />
				<Route path='/home' exact component={UserHome} />
				<Route path='/recipe' component={Recipe} />
				<Route path='/profile' component={Profile} />
			</Switch>
    );
  }
}
