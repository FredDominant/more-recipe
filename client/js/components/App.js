import React from 'react';
import ReactDOM from 'react-dom';

import { Search } from './Search';
import { Navigation } from './Navigation';
import { Body } from './Body';

 export class App extends React.Component {
  render() {
    return (
      <div>
					<Body />
      </div>
    );
  }
};
