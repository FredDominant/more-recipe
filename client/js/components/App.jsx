import React from 'react';

import Body from './Body';
import Errorboundary from '../components/Errorboundary';
import Navbar from './Navbar';

const App = () => (
  <div>
    <Navbar>
      <Errorboundary >
        <Body />
      </Errorboundary>
    </Navbar>
  </div>
);
export default App;
