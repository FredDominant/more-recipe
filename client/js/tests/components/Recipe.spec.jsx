import React from 'react';
import { shallow } from 'enzyme';

import { Recipe } from '../../components/Recipe';


describe('SearchPage component should', () => {
  it('render correctly', () => {
    const props = {
      getRecipeDetails: jest.fn(),
      upvoteRecipe: jest.fn(),
      downvoteRecipe: jest.fn(),
      favourite: jest.fn(),
      recipe: {},
      authenticated: false,
      fetching: false,
      favourited: false,
      match: {
        params: {
          recipeId: 1
        }
      },
    };
    const shallowWrapper = shallow(<Recipe {...props} />);
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('render loading gif is recipe is being fetched', () => {
    const props = {
      getRecipeDetails: jest.fn(),
      upvoteRecipe: jest.fn(),
      downvoteRecipe: jest.fn(),
      favourite: jest.fn(),
      recipe: {},
      authenticated: false,
      fetching: true,
      favourited: false,
      match: {
        params: {
          recipeId: 1
        }
      },
    };
    const shallowWrapper = shallow(<Recipe {...props} />);
    shallowWrapper.setState({
      recipe: {
        name: 'name',
        description: 'description',
        ingredients: 'ingredient, ingredient',
        directions: 'direction'
      }
    });
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('render loading gif is recipe is being fetched', () => {
    const props = {
      getRecipeDetails: jest.fn(),
      upvoteRecipe: jest.fn(),
      downvoteRecipe: jest.fn(),
      favourite: jest.fn(),
      recipe: {},
      authenticated: false,
      fetching: false,
      favourited: false,
      match: {
        params: {
          recipeId: 1
        }
      },
    };
    const shallowWrapper = shallow(<Recipe {...props} />);
    shallowWrapper.setState({
      recipe: {
        id: 1,
        name: 'name',
        description: 'description',
        ingredients: 'ingredient, ingredient',
        directions: 'direction'
      }
    });
    expect(shallowWrapper).toMatchSnapshot();
  });
});

describe('handleUpvote()', () => {
  it('should work for authenticated users', () => {
    const props = {
      getRecipeDetails: jest.fn(),
      upvoteRecipe: jest.fn(),
      downvoteRecipe: jest.fn(),
      favourite: jest.fn(),
      recipe: {
        id: 1,
        name: 'name',
        description: 'description',
        ingredients: 'ingredient, ingredient',
        directions: 'direction'
      },
      authenticated: true,
      fetching: false,
      favourited: false,
      match: {
        params: {
          recipeId: 1
        }
      },
    };
    const shallowWrapper = shallow(<Recipe {...props} />);
    shallowWrapper.setState({
      recipe: {
        id: 1,
        name: 'name',
        description: 'description',
        ingredients: 'ingredient, ingredient',
        directions: 'direction'
      }
    });
    const upvoteButton = shallowWrapper.find('#upvote');
    upvoteButton.simulate('click');
    expect(shallowWrapper.instance().props.upvoteRecipe).toBeCalled();
  });
});

describe('handleDownvote()', () => {
  it('should work for authenticated users', () => {
    const props = {
      getRecipeDetails: jest.fn(),
      upvoteRecipe: jest.fn(),
      downvoteRecipe: jest.fn(),
      favourite: jest.fn(),
      recipe: {
        id: 1,
        name: 'name',
        description: 'description',
        ingredients: 'ingredient, ingredient',
        directions: 'direction'
      },
      authenticated: true,
      fetching: false,
      favourited: false,
      match: {
        params: {
          recipeId: 1
        }
      },
    };
    const shallowWrapper = shallow(<Recipe {...props} />);
    shallowWrapper.setState({
      recipe: {
        id: 1,
        name: 'name',
        description: 'description',
        ingredients: 'ingredient, ingredient',
        directions: 'direction'
      }
    });
    const upvoteButton = shallowWrapper.find('#downvote');
    upvoteButton.simulate('click');
    expect(shallowWrapper.instance().props.downvoteRecipe).toBeCalled();
  });
});


describe('handleFavourite()', () => {
  it('should work for authenticated users', () => {
    const props = {
      getRecipeDetails: jest.fn(),
      upvoteRecipe: jest.fn(),
      downvoteRecipe: jest.fn(),
      favourite: jest.fn(),
      recipe: {
        id: 1,
        name: 'name',
        description: 'description',
        ingredients: 'ingredient, ingredient',
        directions: 'direction'
      },
      authenticated: true,
      fetching: false,
      favourited: false,
      match: {
        params: {
          recipeId: 1
        }
      },
    };
    const shallowWrapper = shallow(<Recipe {...props} />);
    shallowWrapper.setState({
      recipe: {
        id: 1,
        name: 'name',
        description: 'description',
        ingredients: 'ingredient, ingredient',
        directions: 'direction'
      }
    });
    const upvoteButton = shallowWrapper.find('#favourite');
    upvoteButton.simulate('click');
    expect(shallowWrapper.instance().props.favourite).toBeCalled();
  });
});


describe('ComponentWillRecieveProps()', () => {
  it('should be called when component is passed in new props', () => {
    const props = {
      getRecipeDetails: jest.fn(),
      upvoteRecipe: jest.fn(),
      downvoteRecipe: jest.fn(),
      favourite: jest.fn(),
      recipe: {
        id: 1,
        name: 'name',
        description: 'description',
        ingredients: 'ingredient, ingredient',
        directions: 'direction',
        Reviews: [],
        User: {}
      },
      authenticated: true,
      fetching: false,
      favourited: false,
      match: {
        params: {
          recipeId: 1
        }
      },
    };
    const wrapper = shallow(<Recipe {...props} />);
    wrapper.instance().componentWillReceiveProps(props);
    expect(wrapper.instance().state.recipe).toEqual(props.recipe);
  });
});
