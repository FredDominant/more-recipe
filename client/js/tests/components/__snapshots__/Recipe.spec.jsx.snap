// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage component should render correctly 1`] = `<NotFoundPage />`;

exports[`SearchPage component should render loading gif is recipe is being fetched 1`] = `
<div
  className="container loading-icon-container"
>
  <div
    className="text-center mt-30 loading-icon"
  >
    <Loading
      color={null}
      singleColor={null}
      size={100}
    />
  </div>
</div>
`;

exports[`SearchPage component should render loading gif is recipe is being fetched 2`] = `
<div>
  <div
    className="container"
  >
    <div
      className="container"
    >
      <br />
      <div
        className="row"
      >
        <div
          className="col-md-6 col-sm-8"
        >
          <div
            className="image-container"
          >
            <img
              alt="name"
              className="img-thumbnail img-fluid"
              height="50"
              width="50"
            />
          </div>
        </div>
        <div
          className="col-md-6 container"
        >
          <h2
            className="text-left recipe-details-name ml-5"
          >
            Name
          </h2>
          <h5
            className="text-left recipe-details-description ml-5"
          >
            Description
          </h5>
          <h5
            className="text-left ml-5 recipe-detail-user"
          >
             
            <small>
               By: 
               
            </small>
          </h5>
          <div
            className="actions ml-5 mb-3"
          >
            <div
              className="container text-left"
            >
              <span
                className="mr-3"
              >
                <i
                  className="far fa-thumbs-up"
                />
                <span>
                   
                </span>
                 
              </span>
              <span
                className="mr-3"
              >
                <i
                  className="far fa-thumbs-down"
                />
                <span>
                   
                </span>
                 
              </span>
              <span>
                <i
                  className="far fa-heart"
                />
                <span>
                   
                </span>
                 
              </span>
            </div>
          </div>
          <br />
          <h5
            className="text-left ml-5 mb-3"
          >
            INGREDIENTS
          </h5>
          <hr />
          <ul
            className="align-left container"
          >
            <li
              className="list-group-item text-left"
              key="ingredient 1"
            >
              Ingredient
            </li>
            <li
              className="list-group-item text-left"
              key="ingredient 2"
            >
               ingredient
            </li>
          </ul>
        </div>
        <div
          className="container mt-3 mb-3 text-left"
        >
          <h5
            className="text-left "
          >
            DIRECTIONS
          </h5>
          <hr />
          Direction
        </div>
      </div>
      <br />
      <div
        className="container-fluid"
        id="review-body"
      >
        <h5
          className="text-left mt-3 mb-3 ml-5"
        >
          REVIEWS
        </h5>
        <div
          className="add-review-form align-center"
        >
          <br />
        </div>
        <div
          className="container review-body"
        >
          <Connect(Reviews)
            recipeId={1}
          />
        </div>
      </div>
    </div>
  </div>
  <Footer />
</div>
`;
